mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -109,7 +114,7 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): Bucketizer = {
     transformSchema(dataset.schema, logging = true)
-    val splits = dataset.stat.approxQuantile($(inputCol),
+    val splits = dataset.select($(inputCol)).na.drop().stat.approxQuantile($(inputCol),
```

Comment:
Yes, I agree that a similar argument applies for approxQuantile methods. I think the most reasonable semantics are to ignore NaN as well.

QuantileSummaries should probably reject insertion of NaN too.

I'd support making that change as well here, and expanding the scope accordingly.